bucket/leaky_bucket_test.go

```diff
@@ -49,12 +51,21 @@ func TestGrabbingADropFromTheBucket(t *testing.T) {
 }
 
 func TestFillingAnAlreadyFullBucket(t *testing.T) {
-	bucketSize := 2
-	bucket := NewLeakyBucket(bucketSize, 0, 0)
+	config := Configuration{Size: 2, Refill: 1, Duration: time.Duration(10) * time.Millisecond}
+	bucket := NewLeakyBucketWithConfiguration(config)
 	bucket.TopUp()
 
 	assert.Equal(t, true, bucket.IsFull())
 	bucket.AddDrops()
 	assert.Equal(t, true, bucket.IsFull())
-	assert.Equal(t, 2, bucket.UnsafeAvailable())
+	assert.Equal(t, 2, bucket.Available())
+}
+
+func BenchmarkBucket(b *testing.B) {
+	config := Configuration{Size: 2, Refill: 1, Duration: time.Duration(1) * time.Millisecond}
```

Comment:
Would've been cool to have this test before and after to see if this new implementation performed better.

But, this code is way cleaner so I'm kinda yoloswag about that.